pageserver/src/repository.rs

```diff
@@ -462,47 +474,82 @@ mod tests {
         let timelineid = ZTimelineId::from_str("11223344556677881122334455667788").unwrap();
         let tline = repo.create_empty_timeline(timelineid, Lsn(0))?;
 
-        let mut snapshot = tline.history()?;
+        let snapshot = tline.history()?;
         assert_eq!(snapshot.lsn(), Lsn(0));
+        let mut snapshot = snapshot.skip_while(|r| match r {
+            Ok(m) => match m.tag {
+                ObjectTag::RelationBuffer(_) => false,
+                _ => true,
+            },
+            _ => true,
+        });
```

Comment:
I think the `Err` case should be let through. Also this seems to be repeated a lot, maybe we should make a helper function.